internal/gatewayapi/globalresources.go

```diff
@@ -28,17 +43,58 @@ func (t *Translator) ProcessGlobalResources(resources *resource.Resources, xdsIR
 
 	for _, xdsIR := range xdsIRs {
 		if containsGlobalRateLimit(xdsIR.HTTP) || containsWasm(xdsIR.HTTP) {
-			xdsIR.GlobalResources = &ir.GlobalResources{}
+			if xdsIR.GlobalResources == nil {
+				xdsIR.GlobalResources = &ir.GlobalResources{}
+			}
 			xdsIR.GlobalResources.EnvoyClientCertificate = &ir.TLSCertificate{
 				Name:        irGlobalConfigName(envoyTLSSecret),
 				Certificate: envoyTLSSecret.Data[corev1.TLSCertKey],
 				PrivateKey:  envoyTLSSecret.Data[corev1.TLSPrivateKeyKey],
 			}
 		}
 	}
+
 	return nil
 }
 
+// processServiceClusterForGateway returns the matching IR key for a gateway and builds a RouteDestination to represent the ProxyServiceCluster
+func (t *Translator) processServiceClusterForGateway(gateway *GatewayContext, resources *resource.Resources) (string, *ir.RouteDestination) {
+	var (
+		labels = make(map[string]string)
+		irKey  string
+	)
+
+	if t.MergeGateways {
```

Comment:
can we use https://github.com/envoyproxy/gateway/blob/dd1c66fb024e1ce76f59fe1480900307e9ae480d/internal/gatewayapi/helpers.go#L244 here, non blocking, can be done in a follow up